### PR TITLE
update api calls in user, notification and offering actions

### PR DESCRIPTION
**Pull request by jenessatan**
Changes to backend code caused the frontend to crash locally because the urls used in the api calls no longer matched. Updated the urls in the offerings, users and notifications actions to use an environment variable to make it easier to adjust the url for development and production environments. 

Server defaults to port 5000 if no PORT env variable is set (previously set to 3001)

### CODE REVIEW

File: src/redux/actions/userActions.js [hunk 1]
```diff
@@ -138,14 +135,11 @@ export const loginUserAsync = (
...
       // authenticate user
-      const authUserResp = await axios.post(
-        "http://localhost:3001/api/users/login",
-        {
-          email,
-          password,
-          isGoogleLogin: user.data.isGoogleUser,
-        }
-      );
+      const authUserResp = await axios.post(`${BASE_URL}/login`, {
```
Comment (abid-salahi) [inline src/redux/actions/userActions.js:138]: im assuming the base url is like http://localhost:3000/api so here you're missing the /users part of the path

File: src/redux/actions/userActions.js [hunk 2]
```diff
@@ -184,11 +178,7 @@ export const authenticateUser = (token) => {
           "auth-token": token,
         },
       };
-      const resp = await axios.post(
-        "http://localhost:3001/api/users/authenticate",
-        {},
-        config
-      );
+      const resp = await axios.post(`${BASE_URL}/authenticate`, {}, config);
```
Comment (abid-salahi) [inline src/redux/actions/userActions.js:181]: im assuming the base url is like http://localhost:3000/api so here you're missing the /users part of the path

File: src/redux/actions/userActions.js [hunk 3]
```diff
@@ -113,9 +112,7 @@ export const loginUserAsync = (
 
     try {
       // check if email exists in db
-      user = await axios.get(
-        `http://localhost:3001/api/users/findUser/${email}`
-      );
+      user = await axios.get(`${BASE_URL}/findUser/${email}`);
```
Comment (abid-salahi) [inline src/redux/actions/userActions.js:115]: im assuming the base url is like http://localhost:3000/api so here you're missing the /users part of the path

File: src/redux/actions/userActions.js [hunk 4]
```diff
@@ -55,19 +57,16 @@ export const registerUserAsync = (user, openedFrom, postingOwnerId) => {
...
-          firstName,
-          lastName,
-          email,
-          postalCode,
-          dateRegistered,
-          password,
-          isGoogleUser,
-        }
-      );
+      const createUserResp = await axios.post(BASE_URL, {
```
Comment (abid-salahi) [inline src/redux/actions/userActions.js:60]: im assuming the base url is like http://localhost:3000/api so here you're missing the /users part of the path

**Review by abid-salahi — APPROVED** [review]

**Review by jacquelineyin — APPROVED** [review]

**Pull request merged**